runtime-modules/stake/src/lib.rs

```diff
@@ -408,6 +408,38 @@ where
         }
     }
 
+    fn slash_immediate(
+        &mut self,
+        slash_amount: Balance,
+        minimum_balance: Balance,
+    ) -> Result<Balance, ImmediateSlashingError> {
+        ensure!(
+            slash_amount > Zero::zero(),
+            ImmediateSlashingError::SlashAmountShouldBeGreaterThanZero
+        );
+
+        match self.staking_status {
+            StakingStatus::Staked(ref mut staked_state) => {
+                // irrespective of wether we are unstaking or not, slash!
+
+                let actually_slashed = staked_state.apply_slash(
+                    Slash {
+                        slash_amount,
+                        // below values are irrelevant
+                        is_active: true,
+                        blocks_remaining_in_active_period_for_slashing: BlockNumber::zero(),
+                        started_at_block: BlockNumber::zero(),
```

Comment:
started_at_block: now() looks better for immediate slash in my opinion. Even if these values are irrelevant.

Reply:
Good observation here.
I think the better solution is to modify the signature of  `fn apply_slash`
from:
```rust
fn apply_slash(
        &mut self,
        slash: Slash<BlockNumber, Balance>,
        minimum_balance: Balance,
    ) -> Balance
```
to:
```
fn apply_slash(
        &mut self,
        slash_amount: Balance,
        minimum_balance: Balance,
    ) -> Balance
```
since it only really needs the `slash_amount` property of `Slash` struct, and we don't need to pass any of those args..

Reply:
This solution is even better.